src/Standards/Squiz/Sniffs/Classes/SelfMemberReferenceSniff.php

```diff
@@ -103,9 +103,10 @@ protected function processTokenWithinScope(File $phpcsFile, $stackPtr, $currScop
             $error = 'Expected 0 spaces before double colon; %s found';
             $data  = [$found];
             $fix   = $phpcsFile->addFixableError($error, $calledClassName, 'SpaceBefore', $data);
+            $i     = $stackPtr;
 
-            if ($fix === true) {
-                $phpcsFile->fixer->replaceToken(($stackPtr - 1), '');
+            while ($fix === true && $tokens[--$i]['code'] === T_WHITESPACE) {
```

Comment:
You should probably wrap this in a changeset to make sure it gets applied in one go.
```php
$phpcsFile->fixer->beginChangeset();
// Various fixes
$phpcsFile->fixer->endChangeset();
```

Reply:
done

Reply:
@ostrolucky Sorry, I wasn't clear enough. It should just wrap the fixer parts (for each of the code blocks were you made changes), like so:
```php
        if ($fix === true) {
            $phpcsFile->fixer->beginChangeset();
            while ($fix === true && $tokens[--$i]['code'] === T_WHITESPACE) {
                $phpcsFile->fixer->replaceToken($i, '');
            }
            $phpcsFile->fixer->endChangeset();
        }
```

Reply:
Why?

Reply:
Because PHPCS will check whether other sniffs try to change the same tokens within a loop and if so, needs to be able to revert the whole set as one. The changeset will then be offered again in the next loop and hopefully be applied without issue.

The way you have set up the changesets now:
* Disregards whether the fixer is even active.
* Combines the changes for two significantly different errors into one changeset.

Does that explain it well enough ?

Reply:
got it